### PR TITLE
Add correction that takes ptll MC/data ratio and applies this as ptVgen corr

### PR DESCRIPTION
**Pull request by kdlong**
* New corrector that can roughly be seen as reweighing the ptV to the data (more accurate would be to use the unfolded results). This is useful for studies of the bias from the ptV modeling
* Small plotting improvements (eoscp for postfix, ignore unc. in ratio if they don't exist in hist, TNPs added to syst map)
* Function to calculate sqrt(SS) of variations (useful for plotting)

### CODE REVIEW

File: utilities/io_tools/output_tools.py [hunk 1]
```diff
@@ -134,7 +134,9 @@ def copy_to_eos(outpath, outfolder=None):
 def write_theory_corr_hist(output_name, process, output_dict, args=None, file_meta_data=None): 
     outname = output_name
     if args is not None and args.postfix is not None:
-        outname += f"_{args.postfix}"
+        # Don't add if the generator name is already modified
+        if args.postfix not in args.generator:
+            outname += f"{args.postfix}"
```
Comment (cippy) [inline utilities/io_tools/output_tools.py:139]: --postfix is actually used to add any string to the output, it should not be linked to what args.generator has

**Review by cippy — COMMENTED** [review]
Looks ok to me from a first check

**Comment by bendavid** [comment]
Does this need something added to wremnants-data? (ie a histogram with the ratios or similar)

**Comment by kdlong** [comment]
> Does this need something added to wremnants-data? (ie a histogram with the ratios or similar)

The correction produced from this should probably be added to wremnants data, but it's not required for this PR to work.

**Comment by kdlong** [comment]
Added to wremnants-data and fixed the issue with the postfix that Marco pointed out (I think in the other PR). Ready to merge from my end

**Comment by davidwalter2** [comment]
The PRValidation looks good, can this be merged?

**Pull request merged**